engine/worker/cmd_update.go

```diff
@@ -31,7 +31,18 @@ func updateCmd(w *currentWorker) func(cmd *cobra.Command, args []string) {
 	return func(cmd *cobra.Command, args []string) {
 		fmt.Printf("CDS Worker version:%s os:%s architecture:%s\n", sdk.VERSION, runtime.GOOS, runtime.GOARCH)
 		var urlBinary string
-		if !FlagBool(cmd, "from-github") {
+		// flag from-github is defined on command update, not on root command
+		if cmd.Flag("from-github") != nil && cmd.Flag("from-github").Value.String() == "true" {
```

Comment:
so should we set the flag on the root command, so the usage would be more clear:
`worker --auto-update --from-github` seems legit

`worker update --from-github` too